### DISCUSSION TITLE
Consider making an IAuthenticator interface for authenticator implementations not tied to MonoBehaviours

### DISCUSSION THREAD

**Discussion by danloughmiller** [FirstGearGames/FishNet#640] [Feature Request]
I wanted to implement an authenticator in a pure C# class however because the Authenticator base class implements MonoBehaviour all possible implementations would need to be MBs also. From what I can see an Authenticator only needs an InitializeOnce and OnRemoteConnection method, and an OnAuthenticationResult event. This should easily be replaceable with a small interface so other implementations can be provided.

**Comment by FirstGearGames** [comment]
I'm not opposed to this. If you can make a PR I'll get it in.
Make sure the **Authenticator** class inherits from your interface, and that your interface has the virtual methods and events.

Marked as Low Priority but it will get in next release if made a PR.

**Comment by FirstGearGames** [comment]
Going to close this out to keep things clean. If you submit a PR I'll see it!
